.github/workflows/dependency-changelog.yml

```diff
@@ -0,0 +1,114 @@
+# **what?**
+# When dependabot create a PR, it always adds the `dependencies` label. This
+# action will add a corresponding changie yaml file to that PR when that label is added.
+# The file is created off a template:
+#
+# kind: Dependencies
+# body: <PR title>
+# time: <current timestamp>
+# custom:
+#   Author: dependabot
+#   Issue: 4904
+#   PR: <PR number>
+#
+# **why?**
+# Automate changelog generation for more visability with automated dependency updates via dependabot.
+
+# **when?**
+# Once a PR is created and it has been correctly labeled with `dependencies`.  The intended use
+# is for the PRs created by dependabot.  You can also manually trigger this by adding the
+# `dependencies` label at any time.
+
+name: Dependency Changelog
+
+on:
+  pull_request:
+    # catch when the PR is opened with the label or when the label is added
+    types: [opened, labeled]
+
+permissions:
+  contents: write
+  pull-requests: read
+
+jobs:
+  dependency_changelog:
+    if: ${{ github.event.label.name == 'dependencies' }}
```

Comment:
If a PR is opened with the label, will this `github.event.label.name ` have a value since the event is a PR open event and not a labeled event? 

If this becomes a snag, I'm ok with this only triggering on labeling right now since for dependencies, these are all auto-opened for us

Reply:
Yes.  It seems that `github.event...` is not always an actual event.  Similar to how `github.event.pull_request.merged` is not an event but an attribute.  Not intuitive.

I did specifically test the case of adding the label before the PR is created and it triggers the workflows as expected and creates a changelog file.